server/data/data-library/src/main/java/org/apache/james/mailrepository/lib/AbstractMailRepository.java

```diff
@@ -157,4 +159,8 @@ public void remove(String key) throws MessagingException {
      */
     protected abstract void internalRemove(String key) throws MessagingException;
 
+    @Override
+    public long size() throws MessagingException {
```

Comment:
same question here

Reply:
Idem, i'm not able able to come up quickly with better implementation for each of the subclass.

For instance, for file implementation, the only available memory datastrcture is a key cache... No really reliable. And there is no theorical diference with the current implementation if we end up reading files.